File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/RelatedCourseDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for related/recommended course.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RelatedCourseDTO {
    private Integer courseId;
    private String title;
    private String headline;
    private String url;
    private String image240x135;
    private String image480x270;
    private String image750x422;
    private String author;
    private Float rating;
    private Integer numReviews;
    private Integer numSubscribers;
    private String contentInfo; // e.g., "2.5 hours"
    private String instructionalLevel;
}

File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/RelatedCourseDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for related/recommended course.
 */
data class RelatedCourseDTO(
    val courseId: Int? = null,
    val title: String? = null,
    val headline: String? = null,
    val url: String? = null,
    val image240x135: String? = null,
    val image480x270: String? = null,
    val image750x422: String? = null,
    val author: String? = null,
    val rating: Float? = null,
    val numReviews: Int? = null,
    val numSubscribers: Int? = null,
    val contentInfo: String? = null, // e.g., "2.5 hours"
    val instructionalLevel: String? = null
)
